test/functional/CacheManagement/src/tests/sharedclasses/TestUtils.java

```diff
@@ -0,0 +1,1551 @@
+			  while (i<lines.length) {
+				  if (inSharedSemaphoresSection) {
+					  if (lines[i].startsWith("0x")) {
+						  String sem = lines[i].substring(11,21).trim();
+	//					  System.out.println("From line '"+lines[i]+"' we have "+sem);
+						  shmSegments.add(sem);
+					  }
+				  }
+				  if (lines[i].indexOf("Semaphore Arrays")!=-1) inSharedSemaphoresSection=true;
+				  if (lines[i].trim().equals("")) inSharedSemaphoresSection=false;
+				  i++;
+			  }
+		  }
+		  return shmSegments;
+		}
+	
+	public static boolean removeSemaphoreForCache(String cacheName) {
+		StringBuffer removeSemCmd = new StringBuffer("ipcrm -s ");
+		String semid = null;
+		RunCommand.execute(getCommand(ListAllCaches), false);
+		String[] lines = RunCommand.lastCommandStderrLines;
+		for (int i = 0; i < lines.length; i++) {
+			if (lines[i].startsWith(cacheName)) {
+				String token[] = lines[i].split("(\\s)+");
+				/* verify cache name */
+				if (token[0].trim().equals(cacheName) == false) {
+					continue;
+				}
+				/* verify non-persistent cache */
+				if ((token[3].trim().equals("no") == false) && (token[3].trim().equals("non-persistent") == false)) {
+					continue;
+				}
+				/* get semaphore */
+				semid = token[7].trim();
+			}
+		}
+			
+		if (semid != null) {
+			removeSemCmd.append(semid);
+			RunCommand.execute(removeSemCmd.toString());
+			return true;
+		} else {
+			return false;
+		}
+	}
+	
+	public static boolean removeSharedMemoryForCache(String cacheName) {
+		StringBuffer removeShmCmd = new StringBuffer("ipcrm -m ");
+		String shmid = null;
+		RunCommand.execute(getCommand(ListAllCaches), false);
+		String[] lines = RunCommand.lastCommandStderrLines;
+		for (int i = 0; i < lines.length; i++) {
+			if (lines[i].startsWith(cacheName)) {
+				String token[] = lines[i].split("(\\s)+");
+				/* verify cache name */
+				if (token[0].trim().equals(cacheName) == false) {
+					continue;
+				}
+				/* verify non-persistent cache */
+				if ((token[3].trim().equals("no") == false) && (token[3].trim().equals("non-persistent") == false)) {
+					continue;
+				}
+				/* get shared memory */
+				shmid = token[6].trim();
+			}
+			
+		}
+		if (shmid != null) {
+			removeShmCmd.append(shmid);
+			RunCommand.execute(removeShmCmd.toString());
+			return true;
+		} else {
+			return false;
+		}
+	}
+	
+	public static String[] getLastCommandStdout() {
+		return RunCommand.lastCommandStdoutLines;
+	}
+	
+	public static String[] getLastCommandStderr() {
+		return RunCommand.lastCommandStderrLines;
+	}
+	
+	public static boolean isOpenJ9() {
+		return System.getProperty("java.vm.vendor").toLowerCase().contains("openj9");
+	}
+	
+	public static String removeJavaSharedResourcesDir(String dir) {
+		if (dir != null && dir.length() > 0) {
+			String tempDir;
+			if (dir.endsWith(java.io.File.separator)) {
+				tempDir = dir.substring(0, dir.length() - 1);
+			}
+			if (tempDir.endsWith(java.io.File.separator + "javasharedresources")) {
```

Comment:
In the case `dir.endsWith(java.io.File.separator)` is false, `tempDir` is uninitialized when reaching this line. It will result in a NULL pointer exception.

Reply:
Updated.